core/types/transaction_marshalling.go

```diff
@@ -134,10 +125,10 @@ func (t *Transaction) UnmarshalJSON(input []byte) error {
 	case AccessListTxType:
 		var itx AccessListTx
 		inner = &itx
-		if dec.AccessList == nil {
-			return errors.New("missing required field 'accessList' in transaction")
+		// Access list is optional for now.
+		if dec.AccessList != nil {
+			itx.Accesses = dec.AccessList
```

Comment:
Not sure about this. Why do we allow a `nil` list on a accesslist-type transaction?

Reply:
This is needed because we have `,omitempty` on `Accesses`. If the transaction has an empty access list, it will not have "accessList" in JSON output, so we can't require it for JSON input.